java/test/jmri/jmrit/display/layoutEditor/LayoutEditorEditorTest.java

```diff
@@ -0,0 +1,348 @@
+package jmri.jmrit.display.layoutEditor;
+
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.GraphicsEnvironment;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JDialog;
+import javax.swing.JMenuBar;
+import javax.swing.MenuElement;
+import jmri.Turnout;
+import jmri.util.JUnitUtil;
+import jmri.util.JmriJFrame;
+import jmri.util.MathUtil;
+import junit.extensions.jfcunit.TestHelper;
+import junit.extensions.jfcunit.eventdata.EventDataConstants;
+import junit.extensions.jfcunit.eventdata.MouseEventData;
+import junit.extensions.jfcunit.finder.AbstractButtonFinder;
+import junit.extensions.jfcunit.finder.DialogFinder;
+import junit.framework.Test;
+import junit.framework.TestSuite;
+import org.junit.Assert;
+import org.netbeans.jemmy.operators.JDialogOperator;
+import org.netbeans.jemmy.operators.JFrameOperator;
+import org.netbeans.jemmy.operators.JMenuBarOperator;
+import org.netbeans.jemmy.operators.JMenuOperator;
+import org.netbeans.jemmy.operators.JPopupMenuOperator;
+import org.netbeans.jemmy.util.NameComponentChooser;
+import org.slf4j.LoggerFactory;
+
+/**
+ * tests for the LayoutEditor Editor view
+ *
+ * @author	George Warner Copyright 2017
+ */
+public class LayoutEditorEditorTest extends jmri.util.SwingTestCase {
```

Comment:
Don't extend jmri.util.SwingTestCase for Jemmy based tests.  jmri.util.SwingTestCase uses JFCUnit, and you can't mix the two.

Reply:
All this code was plagiarized from other sources that uses jemmy… (Mainly because I don't know any of this unit testing stuff); If you can suggest a different test file to plagiarize I'd welcome it.

Reply:
Use the speedometer test as an example

https://github.com/JMRI/JMRI/blob/master/java/test/jmri/jmrit/speedometer/SpeedometerFrameTest.java

All the jemmy code is tucked away here:
https://github.com/JMRI/JMRI/blob/master/java/test/jmri/jmrit/speedometer/SpeedometerScaffold.java

Reply:
btw, where did you borrow that from?  That sounds like a test class that needs work.

Reply:
Probably from one of these:
```
term> find java -type f -name '*Test.java' -exec grep -i -l 'extends jmri.util.SwingTestCase' {} \;

java/test/jmri/jmrit/beantable/LightTableWindowTest.java
java/test/jmri/jmrit/beantable/LRouteTableActionTest.java
java/test/jmri/jmrit/beantable/OBlockTableActionTest.java
java/test/jmri/jmrit/beantable/SensorTableWindowTest.java
java/test/jmri/jmrit/beantable/TurnoutTableWindowTest.java
java/test/jmri/jmrit/catalog/ImageIndexEditorTest.java
java/test/jmri/jmrit/dispatcher/DispatcherFrameTest.java
java/test/jmri/jmrit/display/IconEditorWindowTest.java
java/test/jmri/jmrit/display/layoutEditor/LayoutEditorConnectivityTest.java
java/test/jmri/jmrit/display/layoutEditor/LayoutEditorWindowTest.java
java/test/jmri/jmrit/display/LinkingLabelTest.java
java/test/jmri/jmrit/display/MemoryIconTest.java
java/test/jmri/jmrit/display/MemorySpinnerIconTest.java
java/test/jmri/jmrit/display/PositionableLabelTest.java
java/test/jmri/jmrit/display/RpsPositionIconTest.java
java/test/jmri/jmrit/display/SensorIconWindowTest.java
java/test/jmri/jmrit/display/SignalMastIconTest.java
java/test/jmri/jmrit/display/TurnoutIconTest.java
java/test/jmri/jmrit/display/TurnoutIconWindowTest.java
java/test/jmri/jmrit/logix/LearnWarrantTest.java
java/test/jmri/jmrit/logix/NXFrameTest.java
java/test/jmri/util/swing/multipane/MultiJfcUnitTest.java
java/test/jmri/util/swing/sdi/SdiJfcUnitTest.java

```

Reply:
Simular grep is showing 23 files still using junit.extensions.jfcunit.
I'll put updating the LayoutEditor ones on my TODO list. ;-)